File: Array/zhangjianan/20200805存在重复元素/containsDuplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        int len=nums.length;
        int i;
        for( i=1;i<len;i++){
            int temp=nums[i];
            int j=i-1;
            for( ;j>0&&temp<nums[j];j-- ){
                nums[j+1]=nums[j];
            }
            nums[j+1]=temp;
        }
        for(i=0;i<len-1;i++){
            if(nums[i]==nums[i+1]) return  true;
        }
        return false;
    }
}

File: Array/wuchao/array/8-7 判断数组是否存在重复元素/ContainsDuplicate.cpp
//
// Created by 吴超 on 2020/8/7.
//
#include <unordered_map>
#include <vector>
using namespace std;
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int,int> m;
        for(int item:nums){
            if(m[item]>0) return true;
            m[item]++;
        }
        return false;
    }
};
